### DISCUSSION TITLE
How do fan and dew heater thresholds work?

### DISCUSSION THREAD

**Discussion by bendenajones** [aaronwmorris/indi-allsky#1509] [Q&A] [answered]
I cannot get the dew heater and fan thresholds to work. I am NOT using PWM for either device. For the fan I expect that once the target temp is exceeded that the fan will turned on. Currently, no matter the target temp and sensor used, the fan always stays on. Only way to get it to turn off is to disable thresholds and the default of disable for "enable night". Basically the same for the dew heater; I would expect that once the target temp is near the dew point that it would turn the heater on. But with the threshold enabled, it stays on.

For the given setup I see in the log: 

sensor.check_fan_thresholds() [502]: Fan threshold current: 66.2, target: 80.0, delta: -13.8

and the fan is turned on.

![image](https://github.com/user-attachments/assets/b56eb49a-2a3e-4331-a570-72f38106feb9)

**Comment by aaronwmorris** [comment]
Do you have a diagram of how your system is wired?  Even a simple one would do.

Depending on how things are wired, it may be necessary to invert the output of the GPIO pins.  The default function is the fan/dew heater is activated by setting the pins HIGH, but some wiring configurations would require the pin to be set LOW to activate instead.

**Reply by bendenajones** [reply]
I don't need to invert the output. I want HIGH to equate to ON and vice versa. If I use the gpioset utility all works as expected. When the Allsky is mounted up on its pole, I refer the image output: 

![image](https://github.com/user-attachments/assets/8e6e5e06-5d5b-48cb-900f-10f2c506bbac)

When down in my office - I can use test leds or just physically test the operation of the fan/heater and it corresponds to the image output device state of ON or OFF.

My issue is with the Threshold settings. As stated, disabling threshold setting indi-allsky will turn on/off the devices according to the "Enable Night" or "Enable Day" options per device type - that all works. As soon as I enable Thresholds the devices get turned on permanently - or virtually so as I have not found a combination of temp and target to disable the target device.

How is it supposed to work? If you can point me to some docs (I haven't found any) that explains the expected behavior I would appreciate it.

**Reply by aaronwmorris** [reply]
Sorry, I thought I responded to this, but I guess I did not hit reply.  I will investigate this.

**Comment by aaronwmorris** [comment, marked as answer]
Okay, I was able to run some tests.  The thresholds appear to be working on my side.

First, a note... you do not have to worry about setting the values to 100 to be ON for the standard dew heaters and fans.  Any value above 0 is ON.

The reason you are seeing this behavior is you have the default level of the fan set to 100%.  The note beside the `Low Threshold Difference` makes note of this behavior.  The default setting will used when the value is below the low threshold.  You should set the default level to 0.
